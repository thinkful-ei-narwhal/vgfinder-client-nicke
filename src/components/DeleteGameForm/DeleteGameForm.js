import React, { Component } from "react";
import { Button, Input } from "../Utils/Utils";
import GamesApiService from "./../../services/games-api-service";
import GamesListContext from "../../contexts/GamesListContext";
import Alert from "./../Alert/Alert";
import "./DeleteGameForm.css";

export default class DeleteGameForm extends Component {
  state = {
    error: null,
    deleteTitle: "",
    deleteEnabled: false,
  };

  static contextType = GamesListContext;

  handleErrorClick = () => {
    this.setState({ error: null });
  };

  handleGameExistsChange = (e, gamesList) => {
    this.setState({
      deleteTitle: e.target.value,
      deleteEnabled: gamesList.find(
        (game) => game.title.toLowerCase() === e.target.value.toLowerCase()
      ),
    });
  };

  handleSubmitGameDelete = (ev, gamesList) => {
    ev.preventDefault();
    this.setState({ error: null });

    GamesApiService.deleteGame(
      gamesList.find(
        (game) =>
          game.title.toLowerCase() === this.state.deleteTitle.toLowerCase()
      ).id
    )
      .then((res) => {
        const updatedgamesList = this.context.gamesList.filter(
          (game) =>
            game.title.toLowerCase() !== this.state.deleteTitle.toLowerCase()
        );
        this.setState({
          error: `The game: ${this.state.deleteTitle} has been deleted!`,
          deleteTitle: "",
          deleteEnabled: false,
        });
        this.context.setGamesList(updatedgamesList);
      })
      .catch((err) => {
        this.setState({ error: err.message });
      });
  };

  render() {
    const { error } = this.state;
    const gamesList = this.context.gamesList;
    return (
      <form
        className="DeleteForm"
        onSubmit={(e) => this.handleSubmitGameDelete(e, gamesList)}
      >
        <div role="alert">
          {error && (
            <Alert message={error} handleErrorClick={this.handleErrorClick} />
          )}
        </div>
        <div className="title">
          <label htmlFor="DeleteGameForm_title">Title</label>
          <Input
            required
            name="title"
            id="DeleteGameForm_title"
            value={this.state.deleteTitle}
            onChange={(e) => this.handleGameExistsChange(e, gamesList)}
          ></Input>
        </div>

        <Button type="submit" disabled={!this.state.deleteEnabled}>
          Delete Game
        </Button>
      </form>
    );
  }
}
