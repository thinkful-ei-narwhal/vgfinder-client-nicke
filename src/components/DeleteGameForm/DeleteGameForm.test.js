import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import DeleteGameForm from "./DeleteGameForm";

describe(`DeleteGameForm component`, () => {
  //add to the db and delete
  //Test the failures

  it("renders a form.DeleteGameForm by default", () => {
    const wrapper = shallow(<DeleteGameForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
