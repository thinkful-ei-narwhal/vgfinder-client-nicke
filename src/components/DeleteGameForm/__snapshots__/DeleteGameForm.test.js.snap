// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteGameForm component renders a form.DeleteGameForm by default 1`] = `
<form
  className="DeleteForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="title"
  >
    <label
      htmlFor="DeleteGameForm_title"
    >
      Title
    </label>
    <Input
      id="DeleteGameForm_title"
      name="title"
      onChange={[Function]}
      required={true}
      value=""
    />
  </div>
  <Button
    disabled={true}
    type="submit"
  >
    Delete Game
  </Button>
</form>
`;
